packages/lit-ssr/src/lib/lit-element-renderer.ts

```diff
@@ -55,6 +55,11 @@ export class LitElementRenderer extends ElementRenderer {
       }
       yield '</style>';
     }
+    // Call LitElement's `willUpdate` method.
+    // Note, this method is required not to use DOM APIs.
+    const changedProperties = new Map();
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    (this.element as any)?.willUpdate(changedProperties);
```

Comment:
Not totally sure it matters since `willUpdate` is not supposed to access DOM, but calling `willUpdate` here will go _after_ attributes are reflected on the server (whereas it goes before attributes are reflected on the client).

While we don't run the user `update` on the server, we do run `ReactiveElement.prototype.update` in order to reflect properties to attributes, and we do that in `connectedCallback` (see L32). It's done in `connectedCallback` and not `renderShadow` because `connectedCallback` is called before closing the tag (`>`), when we can still emit attributes, whereas `renderShadow` is called after the tag is closed when we're ready to emit child content.

TL;DR: seems like this should maybe just go in `connectedCallback` above, before `ReactiveElement.prototype.update()`?

Reply:
Done.